suite-common/wallet-utils/src/solanaStakingUtils.ts

```diff
@@ -0,0 +1,74 @@
+import { LAMPORTS_PER_SOL } from '@solana/web3.js';
+// @ts-expect-error: Could not find a declaration file for module '@everstake/wallet-sdk/solana'.
+import { getDelegations, selectNetwork } from '@everstake/wallet-sdk/solana';
+
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { BigNumber, isArrayMember } from '@trezor/utils';
+import {
+    Account,
+    SolanaStakingAccount,
+    supportedSolanaNetworkSymbols,
+    SupportedSolanaNetworkSymbols,
+} from '@suite-common/wallet-types';
+
+export function isSupportedSolStakingNetworkSymbol(
+    networkSymbol: NetworkSymbol,
+): networkSymbol is SupportedSolanaNetworkSymbols {
+    return isArrayMember(networkSymbol, supportedSolanaNetworkSymbols);
+}
+
+export const formatSolanaStakingAmount = (amount: string | null) => {
+    if (!amount) return null;
+
+    return new BigNumber(amount).div(LAMPORTS_PER_SOL).toFixed(9);
+};
+
+type SolhNetwork = 'devnet' | 'mainnet-beta';
+
+export const getSolNetworkForWalletSdk = (symbol: NetworkSymbol): SolhNetwork => {
+    const solNetworks: { [key in NetworkSymbol]?: SolhNetwork } = {
+        dsol: 'devnet',
+        sol: 'mainnet-beta',
+    };
+
+    return solNetworks[symbol] || solNetworks.sol!;
+};
+
+export const selectSolanaWalletSdkNetwork = (symbol: NetworkSymbol) => {
+    const network = getSolNetworkForWalletSdk(symbol);
+    selectNetwork(network);
+};
+export const calculateTotalStakingBalance = (accounts: SolanaStakingAccount[]) => {
+    if (!accounts?.length) return null;
+
+    const totalAmount = accounts.reduce((acc, solAccount) => {
+        const { account } = solAccount;
+
+        return acc.plus(account?.data.parsed.info.stake.delegation.stake);
```

Comment:
this probably does not include mev kickback and rent. Do we want to count those as well?

Reply:
Yes, this doesn't include either of them for now